### ISSUE TITLE
Show valid attribute locations for warning C0658

### ISSUE THREAD

**Issue by leppie** [dotnet/roslyn#6061]
Warning C0657 shows valid attribute locations when a recognized but invalid location is used.

Warning C0658 should also show valid attribute locations 

Repro: 
````
using System;

[AttributeUsage(AttributeTargets.All)] 
public class All : Attribute {}

public class C {
    [field : All]
    [foo : All]
    public void M() {}
}
````
Following warnings are reported:
```
(6,5,6,10): Warning CS0657: 'field' is not a valid attribute location for this declaration. Valid attribute locations for this declaration are 'method, return'. All attributes in this block will be ignored.
(7,5,7,8): Warning CS0658: 'foo' is not a recognized attribute location. All attributes in this block will be ignored.
```

**Comment by gafter** [comment]
Closed in #6930
